mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveShell.java

```diff
@@ -0,0 +1,179 @@
+import org.apache.hive.service.cli.CLIService;
+import org.apache.hive.service.cli.HiveSQLException;
+import org.apache.hive.service.cli.OperationHandle;
+import org.apache.hive.service.cli.RowSet;
+import org.apache.hive.service.cli.SessionHandle;
+import org.apache.hive.service.cli.session.HiveSession;
+import org.apache.hive.service.server.HiveServer2;
+import org.apache.iceberg.hive.TestHiveMetastore;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+/**
+ * Test class for running HiveQL queries, essentially acting like a Beeline shell in tests.
+ *
+ * It spins up both an HS2 and a Metastore instance to work with. The shell will only accept
+ * queries if it has been previously initialized via {@link #start()}, and a session has been opened via
+ * {@link #openSession()}. Prior to calling {@link #start()}, the shell should first be configured with props that apply
+ * across all test cases by calling {@link #setHiveConfValue(String, String)} ()}. On the other hand, session-level conf
+ * can be applied anytime via {@link #setHiveSessionValue(String, String)} ()}, once we've opened an active session.
+ */
+public class TestHiveShell {
+
+  // We need to use increased pool size in these tests. See: #1620
+  private static final int METASTORE_POOL_SIZE = 15;
+
+  private final TestHiveMetastore metastore;
+  private final HiveServer2 hs2;
+  private final HiveConf hs2Conf;
+  private CLIService client;
+  private HiveSession session;
+  private boolean started;
+
+  public TestHiveShell() {
+    metastore = new TestHiveMetastore();
+    hs2Conf = initializeConf();
+    hs2 = new HiveServer2();
+  }
+
+  public void setHiveConfValue(String key, String value) {
+    Preconditions.checkState(!started, "TestHiveShell has already been started. Cannot set Hive conf anymore.");
+    hs2Conf.verifyAndSet(key, value);
+  }
+
+  public void setHiveSessionValue(String key, String value) {
+    Preconditions.checkState(session != null, "There is no open session for setting variables.");
+    try {
+      session.getSessionConf().set(key, value);
+    } catch (Exception e) {
+      throw new RuntimeException("Unable to set Hive session variable: ", e);
+    }
+  }
+
+  public void start() {
+    metastore.start(METASTORE_POOL_SIZE);
+    hs2Conf.setVar(HiveConf.ConfVars.METASTOREURIS, metastore.hiveConf().getVar(HiveConf.ConfVars.METASTOREURIS));
+    hs2Conf.setVar(HiveConf.ConfVars.METASTOREWAREHOUSE,
+        metastore.hiveConf().getVar(HiveConf.ConfVars.METASTOREWAREHOUSE));
+
+    hs2.init(hs2Conf);
+    hs2.start();
+    client = hs2.getServices().stream()
+            .filter(CLIService.class::isInstance)
+            .findFirst()
+            .map(CLIService.class::cast)
+            .get();
+    started = true;
+  }
+
+  public void stop() {
+    if (client != null) {
+      client.stop();
+    }
+    hs2.stop();
+    metastore.stop();
+    started = false;
+  }
+
+  public TestHiveMetastore getMetastore() {
```

Comment:
Nit: we avoid using `get` because it doesn't add much value. Getter methods are usually just named for the field that is returned.

Reply:
sure, renamed it